java-checks/src/main/java/org/sonar/java/checks/CollectorsToListCheck.java

```diff
@@ -107,6 +107,9 @@ private static boolean isInvariantTypeArgument(MethodInvocationTree collectMetho
     if (streamType.isRawType()) {
       return true;
     }
+    if(collectMethodInvocation.symbolType().typeArguments().isEmpty()){
+      return false;
+    }
     Type collectArgType = collectMethodInvocation.symbolType().typeArguments().get(0);
     Type streamArgType = streamType.typeArguments().get(0);
     return collectArgType.is(streamArgType.fullyQualifiedName());
```

Comment:
Also, in Java 21 we have `List.getFirst()` which we should use as a replacement for `List.get(0)`. Maybe we can shorten the above code even a bit more like this?

```suggestion
    List<Type> typeArguments = collectMethodInvocation.symbolType().typeArguments();
    return !typeArguments.isEmpty() && typeArguments.getFirst().is(streamType.typeArguments().getFirst().fullyQualifiedName());
```